packages/workbox-strategies/src/StrategyHandler.ts

```diff
@@ -556,13 +560,17 @@ class StrategyHandler {
    *
    * Note: any work done after `doneWaiting()` settles should be manually
    * passed to an event's `waitUntil()` method (not this handler's
-   * `waitUntil()` method), otherwise the service worker thread my be killed
+   * `waitUntil()` method), otherwise the service worker thread may be killed
    * prior to your work completing.
    */
   async doneWaiting(): Promise<void> {
-    let promise;
-    while ((promise = this._extendLifetimePromises.shift())) {
-      await promise;
+    while (this._extendLifetimePromises.length) {
+      const promises = this._extendLifetimePromises.splice(0);
+      const result = await allSettled(promises);
```

Comment:
Should this reuse `Promise.allSettled()` if it is available?

```suggestion
      const result = Promise.allSettled ? await Promise.allSettled(promises) : await allSettled(promises);
```